### DISCUSSION TITLE
How to get my bot to announce when I make a social media post?

### DISCUSSION THREAD

**Discussion by camistired** [discordjs/discord.js#6843] [Q&A] [answered]
I'm not super advanced with coding, I know the basics and that's it. I followed a guide so far and am now at the part where I need to actually set up an intent for my bot, but I cannot figure out how to make it so my bot will detect when I make an Instagram post for example. I've googled a bunch of different combinations of my question and I can't find anything on how to set it up because it is just a bunch of people advertising their bots. Any help would be appreciated, if I need to go into more detail I can.

**Comment by almostSouji** [comment, marked as answer]
You would need to check out the API for the respective social media platform. Some allow programmatically querying recent posts (in a polling approach), others let you subscribe to a websocket, and some don't offer any API to provide this feature.

In the latter case you may want to consider scraping the website content itself with tools like [cheerio](https://github.com/cheeriojs/cheerio).

If you need to interact with the website to reach the content in question you may find [puppeteer](https://github.com/puppeteer/puppeteer) or similar tools useful. If you resort to this, however, I'd suggest reading the terms of service of the service you are interacting with in depth, since programmatical interactions are often times not allowed.
